imap/spool.c

```diff
@@ -207,6 +207,30 @@ static int parseheader(struct protstream *fin, FILE *fout,
             break;
 
         case BODY_START:
+            /* leading whitespace might be folded too */
+            if (c == '\r' || c == '\n') {
+                int peek;
+
+                peek = prot_getc(fin);
+
+                if (!skip) buf_appendcstr(&raw, "\r\n");
+
+                if (c == '\r' && peek == '\n') {
+                    c = prot_getc(fin);
+                }
+                else {
+                    c = peek;
```

Comment:
OK, so this allows folded-whitespace to either be CRLF+<WS> or LF+<WS>. Which looks like a sensible thing to do, if somehow a stray newline made it through here. But it should be better included in the test, so at least we can make informed decision if we later decide to change it.

Reply:
Yep, again just using the same logic as BODY.  But actually testing it is a great idea, I'll do that.

Reply:
I've extended test_empty_headers to do the same three checks with each of bare CR and LF